### ISSUE TITLE
File uploaders should check filename with javascript, rather than checking on the backend after upload

### ISSUE THREAD

**Issue by gdevenyi** [aces/Loris#2870]
I have a particularly slow connection, uploading a ZIP file with a typo in the name results in the error getting thrown after the upload rather than as a pre-submit check in Javascript.

**Comment by johnsaigle** [comment]
Could you clarify what you mean by a typo? What was the intended filename and the inputted filename? Which error?

**Comment by gdevenyi** [comment]
When I use the imaging uploader, I specify PSCID, CandID and VisitID which imposes a naming scheme on my zip file of:
```
<PSCID>_<CandID>_Initial_MRI.zip
```

Say I misspell "Initial" to be "Inital" as I did yesterday.

I select the file in my file browser, click upload, the file uploads, then LORIS errors out because the file is named wrong.

I think instead, it would make sense to use the inputted fields and filename to check the filename with javascript before upload. Reduces bandwidth use, etc.

**Comment by johnsaigle** [comment]
You're right, that is annoying and inconsistent with other parts of the codebase. In some places, such as in the `media` uploader, we do have front-end alerts that do this sort of front-end validation. However, it has not yet been fully implemented across LORIS.

In future versions, this behaviour will be more consistent. 

Thanks for bringing this up.
